### ISSUE TITLE
PolyTree logging issue

### ISSUE THREAD

**Issue by ascillitoe** [Effective-Quadratures/equadratures#252]
Using PolyTree with the following parameters returns an error when `logging=True`:

```python
order = 1
max_depth = 3

polytree = PolyTree(splitting_criterion='model_agnostic', max_depth=max_depth, min_samples_leaf=None, 
         k=0.5, order=order, basis='hyperbolic-basis', search='grid',samples=50,
         logging=True, poly_method='least-squares', poly_solver_args=None)
polytree.fit(X_train_eq,y_train.to_numpy())
```

The relevant error output is:
```
in _splitter(node)
    192                                                         # Update best parameters if loss is lower
    193                                                         if loss_split < loss_best:
--> 194                                                                 if self.logging: self.log.append({'event': 'best_split', 'data': {'j_feature':j_feature, 'threshold':threshold, 'loss': loss_split, 'poly_left': poly_left, 'poly_right': poly_right}})
    195                                                                 did_split = True
    196                                                                 loss_best = loss_split

UnboundLocalError: local variable 'poly_left' referenced before assignment
```

**Comment by kenkirito** [comment]
can i work on this issue

**Comment by mHash1m** [comment]
Hi, I'm new to the code base but I could try to resolve this issue if that's okay with you :). @ascillitoe

**Comment by ascillitoe** [comment]
Hi @kenkirito and @mHash1m , welcome! Sorry for the delay getting back to you regarding this. I believe this issue has actually been fixed by @barneyhill in 67d3c4a8a0942b968157d5a034e08e0e4b1e4a10. I have just been slow to merge `feature_polytree` into `develop`. I will close this issue now, apologies for the confusion.

p.s. @mHash1m , I have just seen your [discourse post](https://discourse.equadratures.org/t/google-summer-of-code-2021-projects/108/9) so will get back to you on there re other issues which you're welcome to work on. @kenkirito please do feel free to have a look too :)

**Comment by mHash1m** [comment]
@ascillitoe Awesome. Thanks!
